File: backend/test_click4.py
import asyncio
from playwright.async_api import async_playwright

async def main():
    async with async_playwright() as p:
        browser = await p.chromium.launch(headless=True, channel="msedge")
        context = await browser.new_context()
        page = await context.new_page()

        print("Navigating to home...")
        await page.goto("http://localhost:3000/")
        await page.wait_for_timeout(2000)

        print("Checking if we need to login...")
        login_btn = await page.query_selector('a[href="/login"]')
        if login_btn:
            print("Clicking login link...")
            await login_btn.click()
            await page.wait_for_timeout(2000)
            
            print("Filling login form...")
            await page.fill('input[type="email"]', 'test@test.com')
            await page.fill('input[type="password"]', 'password123')
            await page.click('button[type="submit"]')
            await page.wait_for_timeout(3000)

        print("Navigating to budget...")
        await page.goto("http://localhost:3000/budget")
        await page.wait_for_timeout(3000)

        print("Page URL is now:", page.url)

        btn_grid = await page.query_selector('#budget-view-grid')
        if not btn_grid:
            print("Grid button NOT FOUND! HTML:")
            print(await page.content())
            await browser.close()
            return
            
        print("Clicking grid button...")
        await btn_grid.click()
        await page.wait_for_timeout(1000)
        
        # Take a screenshot to see what it looks like after clicking Grid
        await page.screenshot(path="budget_grid_view_test.png")
        print("Screenshot saved to budget_grid_view_test.png")

        # Check if the class changed
        classes = await btn_grid.get_attribute("class")
        if "bg-primary-500" in classes:
            print("SUCCESS! Grid button turned active.")
        else:
            print("FAIL! Grid button is NOT active. Classes:", classes)
            
        # Check if BudgetTable exists
        table = await page.query_selector('table')
        if table:
            print("BudgetTable is STILL RENDERED!")
        else:
            print("BudgetTable is GONE! (This is correct for grid view)")

        await browser.close()

if __name__ == "__main__":
    asyncio.run(main())

File: backend/test_click4.js
const { chromium } = require('playwright');

(async () => {
  const browser = await chromium.launch({ headless: true });
  const context = await browser.newContext();
  const page = await context.newPage();

  // Inject token
  await page.addInitScript(() => {
    localStorage.setItem('zenith_token', 'dummy-token');
    localStorage.setItem('zenith_user', JSON.stringify({
      id: "test", name: "Test User", email: "test@test.com", role: "user"
    }));
  });

  // Login properly
  await page.goto('http://localhost:3000/');
  await page.click('text="Sign In"');
  await page.waitForTimeout(1000);
  await page.fill('input[type="email"]', 'test@example.com');
  await page.fill('input[type="password"]', 'password123');
  await page.click('button[type="submit"]');
  await page.waitForTimeout(2000);

  await page.goto('http://localhost:3000/budget');
  await page.waitForTimeout(2000);

  const btn = await page.$('#budget-view-grid');
  if (btn) {
    const box = await btn.boundingBox();
    console.log('Button box:', box);
    if (box) {
      const el = await page.evaluate(({x, y}) => {
        const e = document.elementFromPoint(x, y);
        if (!e) return null;
        
        // Find if any parent is AIFloatingWidget or TopNavbar
        let current = e;
        let parents = [];
        while(current && current !== document.documentElement) {
          parents.push(`${current.tagName.toLowerCase()}${current.id ? '#'+current.id : ''}${current.className ? '.'+current.className.split(' ').join('.') : ''}`);
          current = current.parentElement;
        }
        
        return {
          tag: e.tagName,
          id: e.id,
          classes: e.className,
          hierarchy: parents
        };
      }, { x: box.x + box.width / 2, y: box.y + box.height / 2 });
      
      console.log('Element at center:', JSON.stringify(el, null, 2));
    }
  } else {
    console.log('Button not found');
  }

  await browser.close();
})();
